python/benchmark/gen_data_distributed.py

```diff
@@ -502,6 +513,286 @@ def make_regression_udf(iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
+                    data_rvs=np.random.randn,
+                )
+
+                # Separate informative and non-informative columns
+                informative_shuffle_indices = np.arange(orig_cols)
+                informative = sparse_matrix[:, :n_informative]
+
+                if use_cupy:
+                    redundant_mul = np.random.rand(n_informative, redundant_cols)
+                else:
+                    redundant_mul = np.random.rand(n_informative, redundant_cols)
+
+                redundants = informative.dot(redundant_mul)
+                sparse_matrix = sp.sparse.hstack([sparse_matrix, redundants]).tocsr()
+            else:
+                # Generate random sparse matrix without redundant columns directly
+                sparse_matrix = sp.sparse.random(
+                    num_rows_per_partition,
+                    orig_cols,
+                    density=density,
+                    random_state=generator,
+                    format="csr",
+                    dtype=dtype,
+                    data_rvs=np.random.randn,
+                )
+
+            # Shuffle the matrix in scipy matrix with support to indexing
+            if shuffle:
+                sparse_matrix = sparse_matrix[:, col_indices]
+
+            # Transform the sparse matrix into array of sparse vectors
+            indices = range(num_rows_per_partition)
+
+            X = []
+            for i in indices:
+                row = sparse_matrix.getrow(i)
+
+                dense_indices = row.indices
+                dense_values = row.data
+
+                dense = sorted(zip(dense_indices, dense_values))
+                X.append(
+                    {
+                        "type": 0,
+                        "size": cols,
+                        "indices": [x[0] for x in dense],
+                        "values": [x[1] for x in dense],
+                    }
+                )
+
+            # Support parameters and library adaptation
+            if use_cupy:
+                generator_p = cp.random.RandomState(partition_seeds[partition_index])
+                ground_truth_cp = cp.asarray(ground_truth)
+                col_indices_cp = cp.asarray(col_indices)
+            else:
+                generator_p = np.random.RandomState(partition_seeds[partition_index])
+
+            # Label Calculation
+            y = sparse_matrix.dot(ground_truth) + bias
+
+            # Type conversion
+            if use_cupy:
+                y = cp.asarray(y)
+            else:
+                y = np.asarray(y)
+
+            # Random Noise
+            if noise > 0.0:
+                y += generator_p.normal(scale=noise, size=y.shape)
+
+            # Logistric Regression sigmoid and sample
+            if logistic_regression:
+                if use_cupy:
+                    prob = 1 - 1 / (1 + cp.exp(-y))
+                    y = cp.random.binomial(1, prob)
+                else:
+                    prob = 1 - 1 / (1 + np.exp(-y))
```

Comment:
same here.

Reply:
Do we use "prob = 1 / (1 + np.exp(-y))" here to replace "prob = 1 - 1 / (1 + np.exp(-y))" ? Same for line 707.